ilastik/shell/gui/ilastikShell.py

```diff
@@ -160,9 +151,88 @@ def setMemoryBytes(self, bytes):
         self.label.setText("Cached Data: %1.1f MB" % (bytes / (1024.0**2.0)))
 
 
-# ===----------------------------------------------------------------------------------------------------------------===
-# === ProgressDisplayManager                                                                                         ===
-# ===----------------------------------------------------------------------------------------------------------------===
+class LogWindow(QTextEdit):
+    """Window/Widget that shows the log history"""
+
+    def __init__(self, data, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._setup_ui(data)
+
+    def _jump_to_end(self):
+        self.verticalScrollBar().setValue(self.verticalScrollBar().maximum())
+
+    def _setup_ui(self, data):
+        self.setWindowTitle("Log")
+        font = QFont("Monospace")
+        font.setStyleHint(QFont.Monospace)
+        self.setLineWrapMode(QTextEdit.NoWrap)
+        self.setFont(font)
+        self.setMinimumWidth(800)
+        self.setText("\n".join(data))
+        self.setReadOnly(True)
+
+    def showEvent(self, event):
+        super().showEvent(event)
+        self._jump_to_end()
+
+    def appendMessage(self, msg: str):
+        self.append(msg)
+        self._jump_to_end()
+
+
+class LogBar(QLabel):
```

Comment:
I think the UI feature this is trying to provide is notifications, so that's what I'd all it, even if it's currently implemented using a special log level